src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -80,6 +81,11 @@ def create_webapp(resource_group_name, name, plan, runtime=None, startup_file=No
     poller = client.web_apps.create_or_update(resource_group_name, name, webapp_def)
     webapp = LongRunningOperation()(poller)
 
+    if app_settings:
```

Comment:
please avoid separate invocation, rather do it as one shot through `client.web_apps.create_or_update`.  This is to avoid a known racing issue among app-setting reset and scm operations following right after this: `_set_remote_or_local_git`.